plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLanguageClientImpl.kt

```diff
@@ -403,7 +408,37 @@ class AmazonQLanguageClientImpl(private val project: Project) : AmazonQLanguageC
         return CompletableFuture.completedFuture(Unit)
     }
 
+    override fun appendFile(params: FileParams) = refreshVfs(params.path)
+
+    override fun createDirectory(params: FileParams) = refreshVfs(params.path)
+
+    override fun removeFile(params: FileParams) = refreshVfs(params.path)
+
+    override fun writeFile(params: FileParams) = refreshVfs(params.path)
+
+    override fun copyFile(params: CopyFileParams) {
+        refreshVfs(params.oldPath)
+        return refreshVfs(params.newPath)
+    }
+
+    private fun refreshVfs(path: String) {
+        if (Path(path).startsWith(userHomePath)) return
+        try {
+            ApplicationManager.getApplication().invokeLater {
+                VfsUtil.markDirtyAndRefresh(false, true, true, Path(path).toFile())
```

Comment:
async should be fine

Reply:
It does not populate the changes immediately if we do async